File: src/algorithms/tower-of-hanoi.cpp
#include <iostream>

void move(int n, char source, char destination, char auxiliary) {
    if (n == 1) {
        std::cout << "Move disk 1 from " << source << " to " << destination << std::endl;
    } else {
        move(n - 1, source, auxiliary, destination);
        std::cout << "Move disk " << n << " from " << source << " to " << destination << std::endl;
        move(n - 1, auxiliary, destination, source);
    }
}

int main() {
    int num_disks = 3;
    move(num_disks, 'A', 'C', 'B');
    return 0;
}

File: src/algorithms/TowerOfHanoi.java
public class TowerOfHanoi {

  public static void move(
    int n,
    char source,
    char destination,
    char auxiliary
  ) {
    if (n == 1) {
      System.out.println("Move disk 1 from " + source + " to " + destination);
    } else {
      move(n - 1, source, auxiliary, destination);
      System.out.println(
        "Move disk " + n + " from " + source + " to " + destination
      );
      move(n - 1, auxiliary, destination, source);
    }
  }

  public static void main(String[] args) {
    int numDisks = 3;
    move(numDisks, 'A', 'C', 'B');
  }
}
